### ISSUE TITLE
database migration

### ISSUE THREAD

**Issue by bbest** [CalCOFI/scripts#1]
Priority databases to migrate
- [ ] hydrographic
- [ ] larval fish
- [ ] zooplankton: only krill, or all?

**Comment by bbest** [comment]
## Loading spatial data for vector tiles, starting with `sanctuaries`

To get this sanctuaries to load into [tile.calcofi.io](https://tile.calcofi.io) after loading the spatial features into the database:

- load features via R:
  https://github.com/CalCOFI/scripts/blob/28dcc8a9bb483b5d92b8c2eb3ec27f56eb0d34b0/load_sanctuaries.Rmd#L50-L61

- [tile.calcofi.io](https://tile.calcofi.io) 
  ![Screen Shot 2022-04-20 at 1 44 05 PM](https://user-images.githubusercontent.com/2837257/164319779-96586f4b-e192-4ba6-9c83-e6a198ab661b.png)

Ran the following SQL in DBeaver:

```sql
SELECT UpdateGeometrySRID('aoi_fed_sanctuaries','geom',4326);
```

after checking with:

```
SELECT
	nspname AS SCHEMA,
	relname AS TABLE,
	relkind, 
	typname, 
	attname AS geometry_column,
	postgis_typmod_srid (atttypmod) AS srid,
	postgis_typmod_type (atttypmod) AS geometry_type
FROM
	pg_class c
	JOIN pg_namespace n ON (c.relnamespace = n.oid)
	JOIN pg_attribute a ON (a.attrelid = c.oid)
	JOIN pg_type t ON (a.atttypid = t.oid)
WHERE
  relkind IN('r', 'v', 'm') AND 
  typname = 'geometry' AND 
  postgis_typmod_srid (atttypmod) != 0 AND
  relname = 'aoi_fed_sanctuaries';
```

References:

* [pg_tileserv](https://github.com/CrunchyData/pg_tileserv#table-layers)
  A very thin PostGIS-only tile server in Go. Takes in HTTP tile requests, executes SQL, returns MVT **tiles.**
* PostGIS: [UpdateGeometrySRID](https://postgis.net/docs/UpdateGeometrySRID.html)

**Comment by bbest** [comment]
### Progress update (2026-06-07/08)

**Ingest pipeline hardened** (#46 closed): canonical `field_dictionary.csv` + `schema_lint`; per-dataset `questions.csv` rendered in every workflow + `questions_email.qmd`; `dataset_status.csv`; `relationships_cross.csv` + release auto-discovery; calcofi4db `match_by_site_datetime()`/`match_nearest_by_depth()` (#47 closed); `RUNBOOK.md`.

**Datasets ingested & uploaded to `gs://calcofi-db/ingest/`:**
- ✅ euphausiids (#26) — `cce-lter_euphausiids`, 10,150 tows
- ✅ zooplankton tow registry (#27) — `pic_zooplankton`, 99,530 CalCOFI-region tows (biovolume measurement still pending from provider)
- ✅ bird & mammal census (#29; partial #30) — `calcofi_bird_mammal_census` (transect/observation/species/behavior)
- ✅ DIC (#25) — already ingested, re-normalized this session

**Phase-4 schema normalization** (prescriptive, to the canonical dictionary — lat_dec→latitude, lon_dec→longitude, datetime_utc→datetime_start_utc, etc.): ichthyo, bottle, dic re-ingested + uploaded; **ctd-cast in progress**; release re-run + downstream (calcofi4r/int-app/docs) to follow.

**Publish:** ERDDAP workflow built via **EDDTableFromParquetFiles** (`publish_calcofi_to_erddap.qmd` + `libs/erddap.R`) for ctd/casts/dic/euphausiids/zooplankton — pending server deploy (#36/#37/#41).

**Blocked on source acquisition:** phytoplankton #28, ZooDb #31, ZooScan #32, phyllosoma #33, PRODO #34, CUFES #35 (commented per issue).

**Comment by bbest** [comment]
📦 Code pushed: **CalCOFI/workflows#60** (workflows: notebooks, libs, metadata, skills) + **CalCOFI/calcofi4db#2** (match helpers). The referenced files are on those PR branches (data/parquet outputs are uploaded to `gs://calcofi-db/`, not git).

**Comment by bbest** [comment]
## Status vs. Statement of Work (2026-06-08)

**All `**` must-complete INGESTS are done** — released in **v2026.06.08** (44 tables, 133.8M rows; frozen DuckLake on GCS):

| SOW dataset | status |
|---|---|
| Bottle Database | ✅ `calcofi_bottle` |
| Fish Eggs & Larvae | ✅ `swfsc_ichthyo` (incl. cephalopods/inverts folded in) |
| Krill | ✅ `cce-lter_euphausiids` |
| Zooplankton biovolume | ◑ tow registry done (#27); biovolume pending provider |
| CTD Cast Files | ✅ `calcofi_ctd-cast` (ctd_thin headline) |
| Cephalopod | ✅ folded into ichthyo |
| Phytoplankton (Venrick) | ✅ `calcofi_phytoplankton` (#28) |
| DIC | ✅ `calcofi_dic` |
| Lobster phyllosoma | ✅ `calcofi_phyllosoma` |
| CUFES | ✅ `swfsc_cufes` |
| Seabird/Mammal census | ✅ `calcofi_bird_mammal_census` |

**Remaining ingest**: ZooDb #31 + ZooScan #32 (blocked on Ohman/DataZoo portal export), PRODO #34 (nice-to-have), acoustic mammals #30.

**Share (portals)**: CTD/Bottle/DIC + euphausiids/zoo/phyto → **ERDDAP** live (CTD syncing, #36); ichthyo → **OBIS** done. Pending: seabirds #43 / mammals #44 → OBIS, zooplankton #42 / phytoplankton (new) → EDI, MET #45 → ERDDAP.

**Process hardening**: field dictionary + schema-lint, per-dataset `questions.csv`, cross-dataset relationship registry, match helpers, schema site (calcofi.io/schema), apps `ctd-viz` + `datacheck`. Keeping this epic open until ZooDb/ZooScan land.
